### DISCUSSION TITLE
Broadcasting camera video content results in black screen when streaming on Restream.io

### DISCUSSION THREAD

**Discussion by Aniketbokre97** [HaishinKit/HaishinKit.swift#1083] [Q&A]
Im using Haishinkit library for live streaming using MTHKView

Im using this code for attach screen, rtmpStream.attachScreen(ScreenCaptureSession(viewToCapture: view))
Inside the main view, there is one camera view.
I have share the whole screen to the streaming server.
when i start streaming on Restream.io , the camera screen not visible .

Expected output : Stream view as same as iphone view.

![IMG_0065](https://user-images.githubusercontent.com/105269069/201336429-463c879c-489b-484f-875e-ee5cd2e63782.png)
<img width="935" alt="Screenshot 2022-11-10 at 8 39 00 PM" src="https://user-images.githubusercontent.com/105269069/201336441-80d3c7b7-387a-456d-92bb-7ac54b25e144.png">

**Comment by shogo4405** [comment]
ScreenCaptureSession uses the https://developer.apple.com/documentation/uikit/uiview/1622589-drawhierarchy method internally.
This method can't capture a gpu-based content such as AVSampleBufferLayer or AVPlayer video content.
This is specification. I can't fix it. see alos https://github.com/shogo4405/HaishinKit.swift/issues/812
